lib/mapping-transform.js

```diff
@@ -5,34 +5,20 @@ const {types: errorTypes} = require('./errors');
 module.exports = class MappingTransform extends Transform {
 
     constructor(fn) {
-        super({ objectMode: true });
+        super({objectMode: true});
         this._function = fn;
-
     }
 
     _transform(chunk, _, callback) {
-        try {
-            Promise.resolve(this._function(chunk))
-                .then(
-                    this._handleResult(callback),
-                    this._handleError.bind(this)
-                );
-        } catch (err) {
-            Promise.resolve(err).then(this._handleError.bind(this));
-        }
-    }
-
-
-    _handleResult(callback) {
-        return (result) => {
-            callback(null, result);
-        }
-    }
-
-    _handleError(err) {
-        this.emit('error', new RiffError(
-            errorTypes.REQUEST_REPLY_FUNCTION_RUNTIME,
-            err
-        ));
+        Promise.resolve(chunk)
+            .then(value => this._function(value))
+            .then(
+                result => callback(null, result),
+                err => callback(new RiffError(errorTypes.REQUEST_REPLY_FUNCTION_RUNTIME, err))
+            )
+            // allow promise to run to completion before propagating error
+            .catch(err => process.nextTick(() => {
```

Comment:
using `process.nextTick()` pushes the error out of the promise chain, so you no longer see that warning, but doesn't do anything meaningful for actually handling the cause of the error.

```suggestion
            .catch(err => {
                if (this.destroyed) {
                    //  the error is unhandleable
                    // TODO log it?
                    return;
                }
                this.destroy(err);
            });
```

Reply:
With this implementation, the invocation does not stop anymore, so the original problem (`curl` hangs) still occurs.

Reply:
(not saying that the initial `(err) => process.nextTick(() => {throw err})` is the way to go as it crashes the entire process)

Reply:
let's fix the root problem (application error handling/propagation) rather than working around a symptom

Reply:
putting this PR on hold, then